jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebDescriptor.java

```diff
@@ -87,156 +88,10 @@ protected InputSource resolveEntity(String pid, String sid)
             {
                 if (!mapped)
```

Comment:
I'd rather see a solution based on the `Environment`. A `Context` knows what environment it is in - "ee8", "ee9", "ee10" etc. The `WebDescriptor` can have a simple static list of names of resources that match each environment, and do the `registerRedirect` as appropriate.

Reply:
I'll give that a whirl